src/log.js

```diff
@@ -399,34 +399,14 @@ export class Log {
         )
       );
     }
-    if (!shouldBeTrueish) {
-      const message = opt_message || 'Assertion failed';
-      const splitMessage = message.split('%s');
-      const first = splitMessage.shift();
-      let formatted = first;
-      const messageArray = [];
-      let i = 2;
-      pushIfNonEmpty(messageArray, first);
-      while (splitMessage.length > 0) {
-        const nextConstant = splitMessage.shift();
-        const val = arguments[i++];
-        if (val && val.tagName) {
-          firstElement = val;
-        }
-        messageArray.push(val);
-        pushIfNonEmpty(messageArray, nextConstant.trim());
-        formatted += stringOrElementString(val) + nextConstant;
-      }
-      const e = new Error(formatted);
-      e.fromAssert = true;
-      e.associatedElement = firstElement;
-      e.messageArray = messageArray;
-      this.prepareError_(e);
-      // __AMP_REPORT_ERROR is installed globally per window in the entry point.
-      self.__AMP_REPORT_ERROR(e);
-      throw e;
+
+    try {
+      return assertion(this == logs.user ? UserError : Error, shouldBeTrueish);
```

Comment:
Couldn't we import `pureUserAssert`/`pureDevAssert` and avoid this `UserError`/`Error` part? Let pure asserts figure out which error it's throwing.

Reply:
Done.